### DISCUSSION TITLE
Support connection string for redis cluster

### DISCUSSION THREAD

**Discussion by marswong** [redis/go-redis#1730] [Ideas]
Connection string is commonly used to manage a large scale of redis servers, the api of `NewClusterClient` is preferred to be:

```go
rdb := redis.NewClusterClient(
    []string{
        "redis://<user>:<pass>@host1:7000/<db>",
        "redis://<user>:<pass>@host2:7000/<db>",
        "redis://<user>:<pass>@host3:7000/<db>"
    }
)
```

ot just keep compatible with former options:

```go
rdb := redis.NewClusterClient(&redis.ClusterOptions{
    URI: []string{
        "redis://<user>:<pass>@host1:7000/<db>",
        "redis://<user>:<pass>@host2:7000/<db>",
        "redis://<user>:<pass>@host3:7000/<db>"
    },
})
```

**Comment by stephaniehingtgen** [comment]
👋 Hey! I just submitted a [PR](https://github.com/go-redis/redis/pull/1924) around a new function `ParseClusterURLs`, like  `ParseURL` but for cluster mode. You could then use that to get the `ClusterOptions`
